paywall/src/unlock.js/web3Proxy.ts

```diff
@@ -121,6 +121,7 @@ export default function web3Proxy(
   let currentLocks: Locks
   let canUseUserAccounts: boolean = !window.web3
   let hasNativeWeb3Wallet = !!window.web3
+  let showIframeWhenReady = false
```

Comment:
Want to make sure I understand correctly:

- `showIframeWhenReady` is only true when we don't have any locks
- when it is true (i.e., we had not previously received locks) and we receive locks, we show the account iframe

Are there any other cases here?

Reply:
This is almost correct.

`showIframeReady` is *only* set to true when the `PostMessages.SHOW_ACCOUNTS_MODAL` message is sent before the main window receives locks. In this case, we know the user accounts iframe is ready to receive logins or new account creation requests, but we don't yet know if this paywall has erc20 locks on it or not.

Then, when we get the locks, if there are erc20 locks, it checks this flag, and if set to true, it shows the iframe. Come to think of it, it should then reset it. I'll add that line.

Reply:
added

Reply:
Perfect, reshowing the account modal was what I had in mind when asking about other cases